lib/install/rust.js

```diff
@@ -0,0 +1,559 @@
+    resolve({
+      path: dir,
+      cleanup: () => {
+        try {
+          fs.removeSync(dir);
+        } catch (e) {
+          // Swallow errors in removing temporary folder. If the folder was
+          // successfully or unsuccessfully moved, it may not exist at its
+          // location in the temporary directory, but this isn't fatal.
+        }
+      }
+    });
+  });
+}
+
+module.exports.toolchainPath = () => {
+  return new Promise((resolve, reject) => {
+    var sdkPlatformPath = path.join(SDK_PATHS.sdk, getPlatform());
+    var values = fs.readdirSync(sdkPlatformPath);
+
+    for (var i = 0; i < values.length; i++) {
+      if (values[i].match(/^toolchain\-/)) {
+        return resolve(path.join(sdkPlatformPath, values[i]));
+      }
+    }
+    return reject(new Error('No toolchain found.'));
+  });
+};
+
+// Checks is CHECKSUM file in our SDK equals our expected checksum.
+// This will resolve with checking that the SDK exists and matches the checksum.
+module.exports.checkTools = (checksumVerify) => {
+  var dir = path.join(SDK_PATHS.sdk, getPlatform());
+  return new Promise((resolve) => {
+    var checksum = fs.readFileSync(path.join(dir, 'CHECKSUM'), 'utf-8');
+    resolve({
+      exists: true,
+      checked: checksumVerify === checksum,
+      path: dir,
+    });
+  }).catch(() => ({
+    exists: false,
+    checked: false,
+    path: dir,
+  }));
+};
+
+module.exports.checkRustlib = (rustv, checksumVerify) => {
+  var dir = path.join(SDK_PATHS.rustlib, rustv);
+  return new Promise((resolve) => {
+    var checksum = fs.readFileSync(path.join(dir, 'CHECKSUM'), 'utf-8');
+    resolve({
+      exists: true,
+      checked: checksumVerify === checksum,
+      path: dir,
+    });
+  }).catch(() => ({
+    exists: false,
+    checked: false,
+    path: dir,
+  }));
+};
+
+module.exports.installTools = () => {
+  var pkgname = 'Tessel build tools';
+  var url = SDK_URLS[getPlatform()];
+  var checksumVerify = null;
+
+  return downloadString(`${url}.sha256`)
+    .then((checksum) => {
+      checksumVerify = checksum;
+      return exports.checkTools(checksumVerify);
+    })
+    .then((check) => {
+      if (check.exists && check.checked) {
+        log.info(`Latest ${pkgname} already installed.`);
+        return;
+      } else if (!check.exists) {
+        log.info(`Installing ${pkgname}...`);
+      } else {
+        log.info(`Updating ${pkgname}...`);
+      }
+
+      fs.mkdirpSync(path.join(osenv.home(), '.tessel/sdk'));
+      return extractTools(checksumVerify, path.basename(url), download(url));
+    });
+};
+
+module.exports.installRustlib = () => {
+  return exports.rustVersion()
+    .then((rustv) => {
+      var pkgname = `MIPS libstd v${rustv}`;
+      var url = RUST_LIB_TGZ_URL.replace('VERSION', rustv);
+      var checksumVerify;
+
+      return downloadString(url + '.sha256')
+        .catch(() => {
+          throw new Error(`Could not find a MIPS libstd matching your current Rust version (${rustv}).`);
```

Comment:
It's obtuse, but the only consumer of `installRustlib` (module.exports.cargo.install) explains that the next step is to change to a stable Rust and re-run the sdk install step.